### ISSUE TITLE
Error `Environment key "es2021" is unknown` or `0:0  error  Parsing error: Invalid ecmaVersion`

### ISSUE THREAD

**Issue by brachkow** [eslint/eslint#14044]
**Tell us about your environment**

Node version: v12.20.1
npm version: v6.14.11
Local ESLint version: v7.18.0

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
default

**Please show your full configuration:**
<details>
<summary>Configuration</summary>

```js
module.exports = {
  env: {
    commonjs: true,
    es2021: true,
    node: true,
  },
  extends: 'eslint:recommended',
  parserOptions: {
    ecmaVersion: 12,
  },
  rules: {},
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```bash
eslint ./file.js
```

**What did you expect to happen?**
File should be linted.

**What actually happened? Please include the actual, raw output from ESLint.**
Error `Environment key "es2021" is unknown` or `0:0  error  Parsing error: Invalid ecmaVersion` if i remove the key.

**Comment by aladdin-add** [comment]
that should not happen - es2021 is supported in latest eslint

can you see what version of espree you are using? (using `npm ls espree` or `yarn list espree`)

**Comment by mdjermanovic** [comment]
> Error `Environment key "es2021" is unknown` or `0:0 error Parsing error: Invalid ecmaVersion` if i remove the key.

Can you share the whole error output? In particular, what's the ESLint version that appears after "Oops! Something went wrong! :("?

**Comment by brachkow** [comment]
```
Error: .eslintrc.js:
	Environment key "es2021" is unknown
    at /usr/local/lib/node_modules/eslint/lib/shared/config-validator.js:169:19
    at Array.forEach (<anonymous>)
    at validateEnvironment (/usr/local/lib/node_modules/eslint/lib/shared/config-validator.js:163:30)
    at validateConfigArray (/usr/local/lib/node_modules/eslint/lib/shared/config-validator.js:334:9)
    at CascadingConfigArrayFactory._finalizeConfigArray (/usr/local/lib/node_modules/eslint/lib/cli-engine/cascading-config-array-factory.js:417:13)
    at CascadingConfigArrayFactory.getConfigArrayForFile (/usr/local/lib/node_modules/eslint/lib/cli-engine/cascading-config-array-factory.js:271:21)
    at FileEnumerator._iterateFilesWithFile (/usr/local/lib/node_modules/eslint/lib/cli-engine/file-enumerator.js:320:43)
    at FileEnumerator._iterateFiles (/usr/local/lib/node_modules/eslint/lib/cli-engine/file-enumerator.js:301:25)
    at FileEnumerator.iterateFiles (/usr/local/lib/node_modules/eslint/lib/cli-engine/file-enumerator.js:251:59)
    at iterateFiles.next (<anonymous>)
```
and if i remove `es2021`

```
./file.js
  0:0  error  Parsing error: Invalid ecmaVersion

✖ 1 problem (1 error, 0 warnings)
```

**Comment by brachkow** [comment]
> that should not happen - es2021 is supported in latest eslint
> 
> can you see what version of espree you are using? (using `npm ls espree` or `yarn list espree`)

I have `espree@7.3.1`

**Comment by aladdin-add** [comment]
well,  I see! you were using a global-installed eslint - which does not es2021 yet.

you can run `npx eslint ./file.js` to use the local-installed eslint. :)

**Comment by luisfelipesdn12** [comment]
> @aladdin-add 
> well, I see! you were using a global-installed eslint - which does not es2021 yet.
> 
> you can run `npx eslint ./file.js` to use the local-installed eslint. :)

It doesn't worked for me.

```cmd
C:\Users\Usuário\Documents\my\api>yarn lint
yarn run v1.22.5
$ npx eslint ./**
C:\Users\Usuário\Documents\Nooio\api\.eslintrc.json:
        Environment key "es2021" is unknown

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

C:\Users\Usuário\Documents\my\api>yarn list eslint
yarn list v1.22.5
warning Filtering by arguments is deprecated. Please use the pattern option instead.
└─ eslint@5.16.0
Done in 1.31s.
```

**Comment by aladdin-add** [comment]
so your local-installed eslint is v5.16.0; please upgrade eslint to latest, and retry.

**Comment by luisfelipesdn12** [comment]
> @aladdin-add 
> so your local-installed eslint is v5.16.0; please upgrade eslint to latest, and retry.

It worked! Thank you so much for the fast response :)

**Comment by aladdin-add** [comment]
Closing this issue as it looks like the question has been answered. If you still need help, please send
a message to our [mailing list](https://groups.google.com/group/eslint) or [chatroom](https://eslint.org/chat). Thanks!
